doc/contributing.rst

```diff
@@ -2,12 +2,211 @@
 Contributing
 #######################
 
+First of all, thank you for thinking about contributing to PyPSA! 
 
-We strongly welcome anyone interested in contributing to this project.
+We strongly welcome anyone interested in PyPSA and all it's related projects, be it
+with new ideas, suggestions, submitting bug reports or contributing code changes.
 
-Submit pull requests / issues to the `github repository <https://github.com/PyPSA/PyPSA>`_.
+**How to contribute**
 
-Join the `mailing list
-<https://groups.google.com/group/pypsa>`_.
+* :ref:`code-contributions`: Implement new features, fix bugs, or improve the performance.
+* :ref:`documentation-contributions`: Improve the documentation by adding new sections, fixing typos, or clarifying existing content.
+* :ref:`example-contributions`: Showcase your work, which could be useful for others.
 
-See the list of :doc:`developers`.
+**Where to go for help**
+
+* In case of code-related **questions**, please post on `stack overflow <https://stackoverflow.com/questions/tagged/pypsa>`_.
+* For non-programming related and more general questions please refer to the `mailing list <https://groups.google.com/group/pypsa>`_.
+* To **discuss** with other PyPSA users, organise projects, share news, and get in touch with the community you can use the `discord server <https://discord.gg/AnuJBk23FU>`_.
+* For **troubleshooting**, please check the `troubleshooting <https://pypsa.readthedocs.io/en/latest/troubleshooting.html>`_ in the documentation.
+* For **guidelines to contribute** to PyPSA, stay right here.
+
+
+.. _code-contributions:
+
+Code
+=====
+
+**Basic workflow**
+
+#. Clone the repository: ``git clone https://github.com/PyPSA/PyPSA.git``
+#. Install with dependencies in editable mode: ``pip install -e .[dev]``
```

Comment:
can we add pre-commit to the `dev` installations (https://github.com/PyPSA/PyPSA/blob/4fee8b5238cd7dab79f19f300b21377f1f72b27d/pyproject.toml#L45) and add a step to activate pre-commit ie.`pre-commit install`